### PR TITLE
Integrations

### PR DESCRIPTION
**Pull request by emintham**
ALM/Analysis Integration Connectors, ALM/Analysis Integrations, ALM/Analysis Jobs (incl. POST).

### CODE REVIEW

File: source/includes/integration_connectors.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+**`GET /api/v2/connectors/`**
+
+### Query Parameters
+
+The following parameters may be used to filter the ALM connectors.
+
+Parameter | Description
+----------|-----------------------------
+system    | Returns all ALM connectors associated with a particular system.
```
Comment (geoffwhittington) [inline source/includes/integration_connectors.md]: We have `ALM` and `Scanner` connectors

File: source/includes/integration_connectors.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+```
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+    "results": [{
+        "id": 1,
+        "name": "HP Rally Connector",
```
Comment (geoffwhittington) [inline source/includes/integration_connectors.md]: Just `Rally Connector`

File: source/includes/integration_connectors.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+{
+    "results": [{
+        "id": 1,
+        "name": "Fortify Connector",
+        "system": "Fortify"
+    }]
+}
+```
+
+Will return a list of all analysis connectors available.
```
Comment (geoffwhittington) [inline source/includes/integration_connectors.md]: Returns a list of ...

File: source/includes/integration_connectors.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+Content-Type: application/json
+
+{
+    "id": 1,
+    "name": "Fortify Connector",
+    "system": "Fortify"
+}
+```
+
+Will return a specific analysis connector.
```
Comment (geoffwhittington) [inline source/includes/integration_connectors.md]: Returns ...

File: source/includes/integrations.md [hunk 5]
```diff
@@ -0,0 +1,356 @@
+# Integrations
+
+
+## Get All ALM Integrations of All Projects
```
Comment (geoffwhittington) [inline source/includes/integrations.md]: Get all ALM Integrations for all Projects

File: source/includes/integrations.md [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+        "project_id": 2,
+        "system": "Rally",
+        "alias": "Rally Integration",
+        "frequency": "manually",
+        "command": "sync_rally"
+    }]
+}
+```
+
+Will return a list of all ALM integrations associated with all projects.
```
Comment (geoffwhittington) [inline source/includes/integrations.md]: Returns a ...

File: source/includes/integrations.md [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+    "results": [{
+        "id": 1,
+        "system": "WhiteHat",
+        "alias": "WhiteHat Integration",
+        "frequency": "manually",
+        "command": "sync_whitehat",
+        "params": {
+            "asset_name": "WebGoat Java",
+            "sde_project": "WhiteHat",
+            "analysis_api_token": "ab9485b-as6as43a-4acb4f62",
```
Comment (geoffwhittington) [inline source/includes/integrations.md]: Redact this token!

File: source/includes/integrations.md [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+        "project_id": 2,
+        "system": "WhiteHat",
+        "alias": "WhiteHat Integration",
+        "frequency": "manually",
+        "command": "sync_whitehat"
+    }]
+}
+```
+
+Will return a list of all Analysis integrations associated with the project having id "project_id".
```
Comment (geoffwhittington) [inline source/includes/integrations.md]: Returns ...

File: source/includes/integrations.md [hunk 9]
```diff
@@ -0,0 +1,356 @@
...
+
+See the [Include Parameters](#include-parameters) section for more details.
+
+Parameter   | Description
+------------|---------------------
+params      | Includes a params field that specifies the configuration parameters for the analysis integration.
+
+---
+
+## Get All Analysis Integrations of a Specific Project
```
Comment (geoffwhittington) [inline source/includes/integrations.md]: ... for a specific Project

File: source/includes/integrations.md [hunk 10]
```diff
@@ -0,0 +1,356 @@
...
+    }]
+}
+
+```
+
+See the [Include Parameters](#include-parameters) section for more details.
+
+Parameter   | Description
+------------|---------------------
+params      | Includes a params field that specifies the configuration parameters for the analysis integration.
```
Comment (geoffwhittington) [inline source/includes/integrations.md]: From my vantage, the value of this include is if you can tie it to the permission `can_edit_analysis_connections` (I think its called)

File: source/includes/integrations.md (hunk 5 above)
Comment (emintham) [inline source/includes/integrations.md]: Not according to the official style rules though... 'all' would be an indefinite pronoun.

http://titlecapitalization.com/#

Rules
Capitalize the first and the last word.
Capitalize nouns, pronouns, adjectives, verbs, adverbs, and subordinate conjunctions.
Lowercase articles (a, an, the), coordinating conjunctions, and prepositions.
Lowercase the "to" in an infinitive (I want to play guitar).
Paraphrased from The Chicago Manual of Style: 16 ed. paragraph 8.155

File: source/includes/integrations.md (hunk 10 above)
Comment (geoffwhittington) [inline source/includes/integrations.md]: Whats the rationale for making this an include parameter?

File: source/includes/integrations.md (hunk 5 above)
Comment (geoffwhittington) [inline source/includes/integrations.md]: That's a fantastic rebuttal !

I thought this was a sentence, not a title.

File: source/includes/integration_connectors.md [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+        "id": 1,
+        "name": "HP Rally Connector",
+        "system": "Rally"
+    }]
+}
+```
+
+Will return a list of all ALM connectors available.
+
+**`GET /api/v2/connectors/`**
```
Comment (funkaoshi) [inline source/includes/integration_connectors.md]: Are we missing an `alm` at the end of the url.

File: source/includes/jobs.md [hunk 12]
```diff
@@ -0,0 +1,412 @@
...
+----------|-----------------------------
+completed | Returns all ALM jobs that have completed, successfully or otherwise.
+connection| Returns all ALM jobs that correspond to a specific connection.
+manual    | Returns all ALM jobs that were created manually.
+succeeded | Returns all ALM jobs that completed successfully.
+user      | Returns all ALM jobs created by the specified user.
+
+---
+
+### Expand Parameters
```
Comment (funkaoshi) [inline source/includes/jobs.md]: Missing a table of what the expand parameters are?

File: source/includes/integrations.md (hunk 7 above)
Comment (emintham) [inline source/includes/integrations.md]: That's actually an invalid token that I mangled from the original but I can redact that.

File: source/includes/jobs.md [hunk 13]
```diff
@@ -0,0 +1,412 @@
...
+        "connection": {
+            "alias": "Mingle Connection",
+            "frequency": "manually",
+            "id": 23,
+            "system": "Mingle"
+        },
+        "user": 1,
+        "succeeded": true,
+        "completed": true,
+        "automatic": true,
```
Comment (geoffwhittington) [inline source/includes/jobs.md]: `manual`

File: source/includes/jobs.md [hunk 14]
```diff
@@ -0,0 +1,412 @@
...
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+    "results": [{
+        "connection": 23,
+        "user": 1,
+        "succeeded": true,
+        "completed": true,
+        "automatic": true,
```
Comment (geoffwhittington) [inline source/includes/jobs.md]: `manual`

File: source/includes/jobs.md [hunk 15]
```diff
@@ -0,0 +1,412 @@
...
+Content-Type: application/json
+
+
+{
+    "results": [{
+        "connection": 23,
+        "user": 1,
+        "succeeded": true,
+        "completed": true,
+        "automatic": true,
```
Comment (geoffwhittington) [inline source/includes/jobs.md]: `manual`

File: source/includes/jobs.md [hunk 16]
```diff
@@ -0,0 +1,412 @@
...
+
+{
+    "results": [{
+        "id": 21,
+        "project_id": 129,
+        "connection": 23,
+        "user": 1,
+        "succeeded": true,
+        "completed": true,
+        "automatic": true,
```
Comment (geoffwhittington) [inline source/includes/jobs.md]: `manual`

File: source/includes/jobs.md [hunk 17]
```diff
@@ -0,0 +1,412 @@
...
+Content-Type: application/json
+
+{
+    "results": [{
+        "project_id": 129,
+        "connection": 23,
+        "user": 1,
+        "succeeded": true,
+        "completed": true,
+        "automatic": true,
```
Comment (geoffwhittington) [inline source/includes/jobs.md]: `manual`

File: source/includes/jobs.md [hunk 18]
```diff
@@ -0,0 +1,412 @@
...
+        "connection": {
+            "alias": "Mingle Connection",
+            "frequency": "manually",
+            "id": 23,
+            "system": "Mingle"
+        },
+        "user": 1,
+        "succeeded": true,
+        "completed": true,
+        "automatic": true,
```
Comment (geoffwhittington) [inline source/includes/jobs.md]: `manual`

File: source/includes/jobs.md [hunk 19]
```diff
@@ -0,0 +1,412 @@
...
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+    "results": [{
+        "connection": 23,
+        "user": 1,
+        "succeeded": true,
+        "completed": true,
+        "automatic": true,
```
Comment (geoffwhittington) [inline source/includes/jobs.md]: `manual`

File: source/includes/integrations.md (hunk 10 above)
Comment (emintham) [inline source/includes/integrations.md]: I was concerned that returning the params for each of the integrations when requesting the whole list of integrations might be slow (staging's data locally took several seconds to complete this request). I was thinking that the user can request the whole list then request individual ones with the params included if they require that information.

File: source/includes/jobs.md (hunk 12 above)
Comment (emintham) [inline source/includes/jobs.md]: Good catch.

Comment (emintham) [inline source/includes/jobs.md]: Added to below the code.

File: source/includes/jobs.md [hunk 20]
```diff
@@ -0,0 +1,430 @@
...
+
+### Query Parameters
+
+The following parameters may be used to filter the analysis integrations resources in the response. See the [Integrations](#integrations) and [Integration Connectors](#integration-connectors) section for more details about the `connection` parameter.
+
+Parameter | Description
+----------|-----------------------------
+completed | Returns all analysis jobs that have completed, successfully or otherwise.
+connection| Returns all analysis jobs that correspond to a specific connection.
+latest    | Returns only the latest analysis jobs.
```
Comment (emintham) [inline source/includes/jobs.md]: Left intentionally vague until we figure a good definition of 'latest'.

Comment (funkaoshi) [inline source/includes/jobs.md]: "The most recent job for each project"

**Comment by geoffwhittington** [comment]
@emintham is this good to merge?

**Comment by emintham** [comment]
Think so. Unless anyone has other comments?

**Comment by geoffwhittington** [comment]
We can always fix it afterwards

**Comment by emintham** [comment]
OK. Can I trouble you to merge it for me? On the way back from the airport.
Just sent my mum off. Thanks.

**Pull request merged**